### ISSUE TITLE
TypeScript definition for hot option

### ISSUE THREAD

**Issue by benmccann** [svitejs/svite#6]
The type for the `hot` option is defined as below. However, [the readme shows an example of passing `true` or `false`](https://github.com/svitejs/svite/tree/main/packages/vite-plugin-svelte#usage) and [SvelteKit uses `true` as well](https://github.com/sveltejs/kit/blob/master/packages/kit/src/core/dev/index.js#L92), so I think perhaps `true` should be a valid option here? Also, I'd be surprised if all those fields were really required?

```
    hot?: undefined | false | {
        /**
         * Enable state preservation when a component is updated by HMR for every
         * components.
         * @default false
         */
        preserveState: boolean;
        /**
         * If this string appears anywhere in your component's code, then local
         * state won't be preserved, even when noPreserveState is false.
         * @default '\@hmr:reset'
         */
        noPreserveStateKey: string;
        /**
         * If this string appears next to a `let` variable, the value of this
         * variable will be preserved accross HMR updates.
         * @default '\@hmr:keep'
         */
        preserveStateKey: string;
        /**
         * Prevent doing a full reload on next HMR update after fatal error.
         * @default false
         */
        noReload: boolean;
        /**
         * Try to recover after runtime errors in component init.
         * @default true
         */
        optimistic: boolean;
        noDisableCss: boolean;
        injectCss?: boolean;
        cssEjectDelay: number;
        absoluteImports: boolean;
        hotApi?: string;
        adapter?: string;
    };
```

**Comment by dominikg** [comment]
The interfaces need to be cleaned up a bit. hot: true and emitCss: true are going to be defaults that don't need to be specified explicitly. 

vite-plugin-svelte should  'just work' by checking vites mode of operation and configuring compiler, svelte-hmr and vite as needed.

Maybe it could make sense to put vite-plugin-svelte related options into svelte.config.c?js under its own key?

eg
``` 
{
compilerOptions: {...}
emitCss: ...
preprocess: ...
vitePluginSvelte: {
  useAutomaticVitePreprocessor: true
}
}
```

**Comment by dominikg** [comment]
not sure if other tooling in the svelte ecosystem needs or wants to know how vite-plugin-svelte is configured. But this would help keep svelte related config in one file instead of it being split between vite.config.js and svelte.config.js

**Comment by benmccann** [comment]
I'd imagine the config living in `vite.config.js` since it's a vite plugin

**Comment by dominikg** [comment]
fixed in 0.11.0 https://github.com/svitejs/svite/blob/12354401030715f33ca080bc1d26659960c69b97/packages/vite-plugin-svelte/src/utils/options.ts#L188
